tasks/keyInfraSetters.ts

```diff
@@ -35,6 +36,40 @@ task("rotate:IntermediateKey")
     }
   });
 
+task("hd:rotate:IntermediateKey")
+  .addParam("verifiercontract", "Address of the Access Token Verifier Contract")
+  .addParam("newintermediate", "Address of the new intermediate key")
+  .setAction(async function (taskArguments: TaskArguments, { ethers }) {
+    const newIntermediateKey = taskArguments.newintermediate;
+    if (!newIntermediateKey || newIntermediateKey === ethers.constants.AddressZero) {
+      throw new Error("Invalid new intermediate key");
```

Comment:
Is there a specific length to this key? If yes would be great if we also add that on this check

Reply:
Fixed in https://github.com/violetprotocol/ethereum-access-token/pull/33/commits/4e2b573cdcef62759550bdd9f9131d3eaa8e71d9